[exporterhelper] Fix metric name preservation during request splitting

lsj4401 opened the pull request:
#### Description
This PR fixes an issue in exporter/exporterhelper's sending_queue where metric names were lost during metricRequest splitting. Previously, when large batches of metrics were processed, the splitting logic in metric_batch.go could cause the name field of some metrics to disappear. This fix ensures that all metric fields are properly preserved when metricRequest objects are split.

#### Link to tracking issue
Fixes #13236

bogdandrutu reviewed (changes requested):
See comment about new added bug. Please add tests

lsj4401 commented on exporter/exporterhelper/metrics_batch.go:
```diff
@@ -243,13 +244,14 @@ func extractSumDataPoints(srcSum pmetric.Sum, capacity int, sz sizer.MetricsSize
-func extractHistogramDataPoints(srcHistogram pmetric.Histogram, capacity int, sz sizer.MetricsSizer) (pmetric.Metric, int) {
+func extractHistogramDataPoints(srcMetric pmetric.Metric, capacity int, sz sizer.MetricsSizer) (pmetric.Metric, int) {
 	m := pmetric.NewMetric()
+	srcMetric.CopyTo(m)
```
I've considered directly setting name, description, unit, and meta, but I'm concerned that new fields could be missed if they're added later. While I know it's inefficient, I think we should stick to using the copy method as per the existing code. Is there a good way to maintain all other fields while only excluding the datapoint?

lsj4401 commented:
> See comment about new added bug. Please add tests

Added tests to confirm the bug and fixed an issue where bytes were being calculated with missing name, description, and unit fields in existing tests.

jmacd commented on exporter/exporterhelper/metrics_batch.go:
```diff
@@ -177,13 +177,20 @@ func extractMetricDataPoints(srcMetric pmetric.Metric, capacity int, sz sizer.Me
 	case pmetric.MetricTypeSummary:
 		destMetric, removedSize = extractSummaryDataPoints(srcMetric.Summary(), capacity, sz)
 	}
+	destMetric.SetName(srcMetric.Name())
+	destMetric.SetDescription(srcMetric.Description())
+	destMetric.SetUnit(srcMetric.Unit())
```
There is also a Metadata field. Thank you for fixing this!

lsj4401 replied:
Got it, thank you! I missed that. I've added the logic to copy metadata from the source and updated the test code accordingly.

lsj4401 commented:
@bogdandrutu 
Would you mind reviewing this again?

dmathieu commented on .chloggen/fix_preserve-metric-names-request-split.yml:
```diff
@@ -0,0 +1,28 @@
+# Use this changelog template to create an entry for release notes.
```
This file needs to have the extension `.yaml`, not `.yml`.

lsj4401 replied:
@dmathieu 
Thanks for the review. I've updated the file extension to .yaml

dmathieu commented on .chloggen/fix_preserve-metric-names-request-split.yml:
```diff
@@ -0,0 +1,28 @@
+component: exporterhelper
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixes an issue in exporterhelper's sending_queue where metric names were lost during metricRequest splitting.
```
exporterhelper is already specified as the component name. 

```suggestion
note: Preserve all metrics metadata when batch splitting.
```

lsj4401 commented:
@bogdandrutu 
Could you please review this pull request?
